test/e2e/framework/framework.go

```diff
@@ -123,9 +123,18 @@ func (f *Framework) BeforeEach() {
 
 	ginkgo.By("Creating kubernetes clients")
 
+	// We accept either KubeContexts + a single KubeConfig variable, or lots of individual
+	// KubeConfigs
+	Expect(len(TestContext.KubeConfigs) > 0 && len(TestContext.KubeContexts) > 0).NotTo(BeTrue())
+
 	for _, context := range TestContext.KubeContexts {
-		f.ClusterClients = append(f.ClusterClients, f.createKubernetesClient(context))
-		f.SubmarinerClients = append(f.SubmarinerClients, f.createSubmarinerClient(context))
+		f.ClusterClients = append(f.ClusterClients, f.createKubernetesClient(TestContext.KubeConfig, context))
+		f.SubmarinerClients = append(f.SubmarinerClients, f.createSubmarinerClient(TestContext.KubeConfig, context))
```

Comment:
If using multiple kubeconfigs then `TestContext.KubeConfig` may/would be empty in which case it looks like `loadConfig` would fail or am I missing something?

Reply:
Yeah, that's true. I will either add an error on the check on previous lines or also support that case.